pkgs/tools/system/netdata/default.nix

```diff
@@ -60,18 +60,12 @@ stdenv.mkDerivation rec {
     ++ lib.optionals withSsl [ openssl ];
 
   patches = [
+    # Allow ndsudo to use non-hardcoded `PATH`
+    # See https://github.com/netdata/netdata/pull/17377#issuecomment-2183017868
+    #     https://github.com/netdata/netdata/security/advisories/GHSA-pmhq-4cxq-wj93
+    ./patches/ndsudo-fix-path.patch
```

Comment:
Is this a security concern for NixOS? Given that it's rated as 8.8/10 severity risk in the netdata's repo.

Reply:
On NixOS, `ndsudo` currently does not have SUID permission.

Reply:
Сan also try replacing `"/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin"` with the full paths to programs `dmsetup`, `ssacli`, `smartctl`, etc. d.
But this will increase package closure size.

Reply:
That would shift those dependencies from runtime to build time. Given how many collectors there are, I don't think it's a good idea.
I have another change pending (no PR yet) to expose `extraPackages` in `services.netdata` to allow end-users to pass their runtime dependencies.

Reply:
For `ndsudo` to work correctly, will need to grant it the SUID privilege. After this, will need to make sure that only a certain path is available, in which only the necessary programs are indicated.
Example:
```
    char new_path[] = "PATH=/run/netdata/ndsudo/bin";
    putenv(new_path);
```

Reply:
Ah, that's a good idea, thanks! But I think it's out of the scope of this PR. 
I'll work on it in the upcoming PR.